### PR TITLE
[engine] Sync Flutter 3.13.1 source code

### PR DESCRIPTION
**Pull request by JSUYA**
Copy embedder.h and client_wrapper from the Flutter 3.13.1 engine source tree.

engine: https://github.com/flutter-tizen/engine/tree/flutter-3.13.1

### CODE REVIEW

**Comment by JSUYA** [comment]
Changes to SystemChannel.
- [ ] https://github.com/flutter/flutter/pull/96637
- [ ] https://github.com/flutter/flutter/pull/121378
- [ ] https://github.com/flutter/flutter/pull/126075
- [ ] https://github.com/flutter/flutter/pull/122885

**Comment by JSUYA** [comment]
https://github.com/flutter-tizen/embedder/pull/36/commits/b0bd1112ec05f36019d74f8012fceb1864b4d3c9

File: flutter/shell/platform/common/client_wrapper/core_wrapper_files.gni [hunk 1]
```diff
@@ -48,3 +48,7 @@ core_cpp_client_wrapper_sources = get_path_info([
                                                   "standard_codec.cc",
                                                 ],
                                                 "abspath")
+
+# Temporary shim, published for backwards compatibility.
+# See comment in the file for more detail.
+temporary_shim_files = get_path_info([ "engine_method_result.cc" ], "abspath")
```
Comment (swift-kim) [inline flutter/shell/platform/common/client_wrapper/core_wrapper_files.gni:54]: This part and the `engine_method_result.cc` file must be removed. The presence of the file may break C++ build.

File: flutter/shell/platform/common/client_wrapper/publish.gni [hunk 2]
```diff
@@ -72,7 +72,8 @@ template("publish_client_wrapper_core") {
                            ])
     public = core_cpp_client_wrapper_includes
     sources = core_cpp_client_wrapper_sources +
-              core_cpp_client_wrapper_internal_headers + [ _wrapper_readme ]
+              core_cpp_client_wrapper_internal_headers + [ _wrapper_readme ] +
+              temporary_shim_files
```
Comment (swift-kim) [inline flutter/shell/platform/common/client_wrapper/publish.gni:76]: Ditto. Please revert.

File: flutter/shell/platform/common/client_wrapper/core_wrapper_files.gni (hunk 1 above)
Comment (JSUYA) [reply flutter/shell/platform/common/client_wrapper/core_wrapper_files.gni]: Thank you for check. I updated code.

File: flutter/shell/platform/common/client_wrapper/publish.gni (hunk 2 above)
Comment (JSUYA) [reply flutter/shell/platform/common/client_wrapper/publish.gni]: done

**Review by swift-kim — APPROVED** [review]

**Pull request merged**